test/functional/page_objects/visual_builder_page.ts

```diff
@@ -838,22 +846,27 @@ export class VisualBuilderPageObject extends FtrService {
     await this.setMetricsGroupBy('terms');
     await this.common.sleep(1000);
     const byField = await this.testSubjects.find('groupByField');
-    await this.retry.try(async () => {
-      await this.comboBox.setElement(byField, field);
-    });
-
+    await this.comboBox.setElement(byField, field);
+    const isSelected = await this.comboBox.isOptionSelected(byField, field);
+    expect(isSelected).to.be(true);
     await this.setMetricsGroupByFiltering(filtering.include, filtering.exclude);
   }
 
   public async setAnotherGroupByTermsField(field: string) {
-    const fieldSelectAddButtons = await this.testSubjects.findAll('fieldSelectItemAddBtn');
-    await fieldSelectAddButtons[fieldSelectAddButtons.length - 1].click();
+    // Using xpath locator to find the last element
+    const fieldSelectAddButtonLast = await this.find.byXPath(
+      `(//*[@data-test-subj='fieldSelectItemAddBtn'])[last()]`
+    );
+    // In case of StaleElementReferenceError 'browser' service will try to find element again
+    await fieldSelectAddButtonLast.click();
     await this.common.sleep(2000);
-    const byFields = await this.testSubjects.findAll('fieldSelectItem');
-    const selectedByField = byFields[byFields.length - 1];
-    await this.retry.try(async () => {
-      await this.comboBox.setElement(selectedByField, field);
-    });
```

Comment:
My guess retry won't help on error, since it won't clean the element state. Hopefully with main PR change this code becomes more stable.